python/federatedml/linear_model/logistic_regression/hetero_sshe_logistic_regression/hetero_lr_guest.py

```diff
@@ -0,0 +1,259 @@
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import functools
+import operator
+
+import numpy as np
+
+from federatedml.linear_model.linear_model_weight import LinearModelWeights
+from federatedml.linear_model.logistic_regression.hetero_sshe_logistic_regression.hetero_lr_base import HeteroLRBase
+from federatedml.optim import activation
+from federatedml.optim.convergence import converge_func_factory
+from federatedml.protobuf.generated import lr_model_param_pb2
+from federatedml.secureprotol.spdz.tensor import fixedpoint_numpy, fixedpoint_table
+from federatedml.util import LOGGER, consts
+from federatedml.util import fate_operator
+from federatedml.util.io_check import assert_io_num_rows_equal
+
+
+class HeteroLRGuest(HeteroLRBase):
+
+    def __init__(self):
+        super().__init__()
+        self.encrypted_error = None
+        self.encrypted_wx = None
+
+    def _init_model(self, params):
+        super()._init_model(params)
+        if not self.is_respectively_reviewed:
+            self.converge_func = converge_func_factory("weight_diff", params.tol)
+
+    def transfer_pubkey(self):
+        public_key = self.cipher.public_key
+        self.transfer_variable.pubkey.remote(public_key, role=consts.HOST, suffix=("guest_pubkey",))
+        remote_pubkey = self.transfer_variable.pubkey.get(role=consts.HOST, idx=0,
+                                                          suffix=("host_pubkey",))
+        return remote_pubkey
+
+    def cal_prediction(self, w_self, w_remote, features, spdz, suffix):
+        z1 = features.dot_array(w_self.value, fit_intercept=self.fit_intercept)
+
+        za_suffix = ("za",) + suffix
+        self.secure_matrix_mul_active(w_remote, cipher=self.cipher, suffix=za_suffix)
+        za_share = self.received_share_matrix(self.cipher, q_field=self.fixpoint_encoder.n,
+                                              encoder=self.fixpoint_encoder, suffix=za_suffix)
+        zb_share = self.secure_matrix_mul_passive(features,
+                                                  suffix=("zb",) + suffix)
+        z = z1 + za_share + zb_share
+
+        # z = z.convert_to_array_tensor()
+        # new_w = z.reconstruct_unilateral(tensor_name=f"z_{self.n_iter_}")
+        # raise ValueError(f"reconstructed z: {new_w}")
+
+        # z_square = z * z
+        # z_cube = z_square * z
+
+        remote_z, remote_z_square, remote_z_cube = self.share_z(suffix=suffix)
+
+        complete_z = remote_z + z
+        # complete_z_cube = remote_z_cube + remote_z_square * z * 3 + remote_z * z_square * 3 + z_cube
+        # sigmoid_z = complete_z * 0.197 - complete_z_cube * 0.004 + 0.5
+        sigmoid_z = complete_z * 0.2 + 0.5
```

Comment:
shouldn't it be 0.25*z, instead of 0.2*z?

Reply:
Thanks for point out. But this pr is going to be abandoned. Thank you anyway.